tests/utils/test_rest_utils.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+import mock
+import pytest
+import requests
+
+from databricks_cli.configure import provider
+from databricks_cli.configure.provider import DatabricksConfig
+from mlflow.utils import rest_utils
+
+
+def _mock_profile(expected_profile, config):
```

Comment:
It could be nice to use `mock.patch.object` (e.g. as we do [here](https://github.com/databricks/mlflow/blob/c7633f228b5527603503fba71f9618898d8c9261/tests/projects/test_databricks.py#L60)) to ensure we clean up / reset `provider` after each test, maybe we could do something like:

```
def _mock_profile(expected_profile, config):
    def mock_get_config_for_profile(profile):
            assert profile == expected_profile
            return config
        provider.get_config_for_profile = mock_get_config_for_profile
    return mock.patch.object(provider, "get_config_for_profile", new=mock_get_config_for_profile)


def test_databricks_params_token():
    with _mock_profile(expected_profile="...", config="..."):
        # perform test, at end of with block `provider` will be reset
```

Reply:
It's so complicated in comparison! I understand how/why the current one works, and if tests fail they fail. I would rather not change unless the test becomes more complicated such that a more compelling reason forms.

Reply:
Yeah agreed that it's more complicated, fine with leaving it as is - also agree that if overriding `provider.get_config_for_profile` becomes a problem it'll likely manifest via another test (e.g. one that calls `provider.get_config_for_profile` in another suite) failing & we can address it then.

Reply:
Oh, I'm pretty sure these suites are run in different processes! One suite cannot affect another, right?